core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -225,6 +219,8 @@ public LogicalPlan visitFilter(Filter node, AnalysisContext context) {
     LogicalPlan child = node.getChild().get(0).accept(this, context);
     Expression condition = expressionAnalyzer.analyze(node.getCondition(), context);
 
+    OpenSearchFunctions.validateFieldList((FunctionExpression)condition, context);
```

Comment:
Trying to understand the cause to see if this can be fixed in consistent Analyzer architecture rather than injecting special logic here. As I understand, the root cause is in `ExpressionAnalyzer`, we didn't continue analyzing field name?

```
  @Override
  public Expression visitRelevanceFieldList(RelevanceFieldList node, AnalysisContext context) {
    return new LiteralExpression(ExprValueUtils.tupleValue(
        ImmutableMap.copyOf(node.getFieldList())));
  }
```

If so, I wonder can this be fixed simply in `ExpressionAnalyzer`?

Reply:
How about this approach?

We add `validateParameters(TypedEnvironment ...)` to `FunctionExpression` interface. By default it always passes.

In [OpenSearchFunctions.OpenSearchFunction](https://github.com/opensearch-project/sql/blob/f8c58f1283868bd5fb15639a382b949c3e50c7db/core/src/main/java/org/opensearch/sql/expression/function/OpenSearchFunctions.java#L82) we implement `validateParameters` to do the current check.

Reply:
@forestmvey @MaxKsyunz Thanks for the idea! I think what Max posted better align with our goal in https://github.com/opensearch-project/sql/issues/811. As long as we can put all OpenSearch function logic together, it would be much easier to move them to `opensearch` module and register dynamically as what we did for Prometheus function here: https://github.com/opensearch-project/sql/blob/2.x/prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusStorageEngine.java#L38

Reply:
@dai-chen I have resolved this fix under [9f5304c166764a1f1b44300606e97fe3bd0117b3](https://github.com/opensearch-project/sql/pull/1067/commits/9f5304c166764a1f1b44300606e97fe3bd0117b3)